### ISSUE TITLE
CardReader battery level seems to be out-dated

### ISSUE THREAD

**Issue by malinajirka** [woocommerce/woocommerce-android#4952]
When a reader is discovered and a software update started, it seems the app shows battery level from the time when the reader was discovered instead of the current value. Are we caching the value or is the out-dated value provided by the SDK? If the out-dated value is provided by the SDK -> report an issue on their GitHub.

**Comment by malinajirka** [comment]
This is an expected behavior. Release of an improvement is estimated for an end of Q1 2022. pdi1mn-3k-p2

**Comment by malinajirka** [comment]
This is an issue in the Stripe SDK.

**Comment by malinajirka** [comment]
Reopening as this issue [has been fixed in version 2.7.0. on 2/28/2022](https://github.com/stripe/stripe-terminal-android/blob/master/CHANGELOG.md).

**Comment by malinajirka** [comment]
The new version of the Stripe Terminal SDK depends on Kotlin 1.6+. Putting this task on hold until we update AGP and Kotlin.

**Comment by kidinov** [comment]
@malinajirka do you know how we can test if this was fixed with 2.9.0?

**Comment by malinajirka** [comment]
@kidinov My understanding is that we need to modify the code - the SDK contains a new ReaderListener callback:
`New: Added onBatteryLevelUpdate callback in ReaderListener both for Bluetooth and USB readers when connected. It reports battery info for every 10 minutes. See https://github.com/stripe/stripe-terminal-android/issues/199`.

The last time I was testing this, I started the payment flow and left the card in the reader for 30+ minutes which drained some battery but the app would keep showing the original battery level.
